client/app/components/navbar.tsx

```diff
@@ -205,27 +267,87 @@ function BottomNav() {
+	const [isDialogOpen, setIsDialogOpen] = React.useState(false);
+
+	const handleLogoutClick = (e: React.MouseEvent) => {
+		e.preventDefault();
+		setIsDialogOpen(true);
+	};
+
+	const handleConfirmLogout = () => {
+		setIsDialogOpen(false);
+	};
+
+	const handleCancelLogout = () => {
+		setIsDialogOpen(false);
+	};
+
 	return (
-		<ul className="">
-			{links.map((link) => (
-				<li key={link.href}>
-					<NavLink
-						to={link.href}
-						className={({ isActive }) =>
-							clsx(
-								"px-2 py-1 hover:bg-zinc-100 dark:hover:bg-neutral-800 rounded-full font-medium flex items-center gap-2 transition-[background] duration-200",
-								{
-									"!bg-zinc-200 !dark:bg-neutral-800": isActive,
-									"text-secondary": !isActive,
-								},
-							)
-						}
+		<div className="flex flex-col h-full justify-between">
+			<ul className="flex-grow">
+				{links.map((link) => (
+					<li key={link.href}>
+						<NavLink
+							to={link.href}
+							className={({ isActive }) =>
+								clsx(
+									"px-2 py-1 hover:bg-zinc-100 dark:hover:bg-neutral-800 rounded-full font-medium flex items-center gap-2 transition-[background] duration-200",
+									{
+										"!bg-zinc-200 !dark:bg-neutral-800": isActive,
+										"text-secondary": !isActive,
+									},
+								)
+							}
+						>
+							<div className={clsx("opacity-70", link.icon)} /> {link.title}
+						</NavLink>
+					</li>
+				))}
+			</ul>
+			{Boolean(user) && (
+				<>
+					<div
+						className="px-2 py-1 hover:bg-zinc-100 dark:hover:bg-neutral-800 rounded-full font-medium flex items-center gap-2 transition-[background] duration-200 group cursor-pointer"
+						onClick={handleLogoutClick}
 					>
-						<div className={clsx("opacity-70", link.icon)} /> {link.title}
-					</NavLink>
-				</li>
-			))}
-		</ul>
+						<div className="i-lucide-arrow-left-circle bg-red-600 opacity-70 " />
+						<span className="text-secondary group-hover:text-dark !dark:group-hover:text-white">
+							Logout
+						</span>
+					</div>
+
+					<Dialog open={isDialogOpen} onClose={handleCancelLogout}>
+						<div className="p-4">
+							<h2 className="text-base font-semibold flex items-center gap-1">
+								Logout{" "}
+							</h2>
+							<p className="mt-2">Are you sure you want to log out?</p>
+							<div className="flex justify-end mt-4 gap-2">
+								<Button
+									className=" !text-black px-2 py-1 text-sm bg-gray-200 dark:bg-neutral-800 text-gray-800 !dark:text-white"
+									onClick={handleCancelLogout}
+								>
+									Cancel
+								</Button>
+								<Form
+									action="logout"
+									method="post"
+									className=" transition-[background] duration-200 group cursor-pointer"
+								>
+									<Button
+										onClick={handleConfirmLogout}
+										className="text-white text-sm !bg-red-500"
+									>
+										Continue
+									</Button>
+								</Form>
+							</div>
+						</div>
+					</Dialog>
+				</>
+			)}
+		</div>
```

Comment:
This is a duplicate of the dialog in navbar. DRY it

Reply:
Solved now. Created a log-out modal component for that